### PR TITLE
Handle big osm2streets changes

### PR DESCRIPTION
**Pull request by dabreegster**
@BudgieInWA and I have been making big changes to osm2streets recently. This PR aims to get A/B Street caught up.

Some behavioral changes / important notes:
- Saved edits reference OSM IDs right now. Now some roads and intersections explicitly track multiple OSM IDs per object, reflecting the reality of the operations happening in osm2streets. This could cause confusing things to happen when people edit near consolidated roads or intersections. The saved edits need a massive overhaul anyway to be robust against this sort of thing, but for now, I'm still punting on #995.
- I'm getting rid of the import option that randomly fills out street parking. If people using maps where parking isn't explicitly tagged yet want to see the parking show up in A/B Street, there are now some tools like https://github.com/zlant/parking-lanes and https://a-b-street.github.io/docs/software/parking_mapper.html that can help
- Minor: osm2streets now uses "map edge" instead of "border", but I'm not changing everything here yet
- No more inserting synthetic OSM tags everywhere for tracking OSM IDs (an old hack used in debug mode). Also no more `ENDPT_FWD`, meaning some turn restrictions could be applied in a bit of a funny way. All of that handling is pretty ad-hoc right now anyway, and we're iterating in osm2streets on doing it properly.
- I'm not attempting to plumb through multiple OSM IDs into `map_model` yet. It has downstream consequences in too many different UIs to bother right now.

This PR is a perfect example of why splitting osm2streets into its own repo has been valuable. The A/B Street codebase is massive and has lots of legacy-ish apps that take more effort to maintain. We've been able to iterate in osm2streets much faster by decoupling when the code here has to get updated. And we're also detached from weird old technical decisions made here.

But the medium-term picture I want to drive towards is making `map_model` just internally store/wrap an `osm2streets::StreetNetwork`. That'll deduplicate some logic and make keeping this repo up-to-date easier. I'm not sure how to do that kind of thing yet. Maybe the development flow will eventually be that experiments start here and, once a bit stable, move into the osm2streets repo, likely as additional crates. 4 examples offhand:

- rendering lane markings in detail (some of that logic is already in osm2streets, expressed as GeoJSON)
- LTN blockfinding. #1024 has big ideas for generalizing this to planar graphs and making it more robust
- Pathfinding, with potentially fancy options for customizing preferences, and optional contraction hierarchy engine (#555 for some improvements)
- Managing map edits (#995 for improvements)

In other words, the current implementation of many of these topics here is not ideal. But it feels like we've got experience to take another stab and do some things over in a cleaner way.

# TODO

I need to test this PR a bunch more, and I already know a few more things needed before regenerating all data...

### CODE REVIEW

**Comment by BudgieInWA** [comment]
Embedding an `osm2streets::StreetNetwork` in `map_model` seems sensible to me. I'm already conceptualising the API in that way, so I'm happy for that use-case to influence the design and development.

I also like the sound of building features - like your listed examples - in this repo, on top of StreetNetwork and extracting the generic form out into `osm2streets` later.

**Comment by dabreegster** [comment]
Good enough to plow forward. One of the Bristol LTN consultation boundaries breaks, but I'm not entirely sure why -- it's a problem in 0.3.37. Since the consultation is over now, it matters less. Also, we could just keep Commonplace pinned to an older version.
